From 689a530d5b38578a0be2a3280611b226a1345e98 Mon Sep 17 00:00:00 2001
From: Daniel Rosenberg <drosen@google.com>
Date: Tue, 13 Jan 2015 16:38:10 -0800
Subject: [PATCH 509/634] cpufreq: stats: hold reference on global cpufreq

As in f8ad216f37e6a7b07d5f6982a25c426c72f661f6,
cpufreq stat must grab a reference to the global
cpufreq kobject in order to use it. Otherwise,
the kobj->sd will be NULL, and trigger a BUG in
sysfs_create_file.

Change-Id: If7d1c13c6ebb1678e1cd33d46eba04a41a0a796d
Signed-off-by: Daniel Rosenberg <drosen@google.com>
---
 drivers/cpufreq/cpufreq_stats.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/cpufreq/cpufreq_stats.c b/drivers/cpufreq/cpufreq_stats.c
index 399bcbf..d3eafcb 100644
--- a/drivers/cpufreq/cpufreq_stats.c
+++ b/drivers/cpufreq/cpufreq_stats.c
@@ -590,6 +590,7 @@ static int __init cpufreq_stats_init(void)
 	}
 
 	create_all_freq_table();
+	WARN_ON(cpufreq_get_global_kobject());
 	ret = sysfs_create_file(cpufreq_global_kobject,
 			&_attr_all_time_in_state.attr);
 	if (ret)
@@ -608,6 +609,7 @@ static void __exit cpufreq_stats_exit(void)
 	for_each_online_cpu(cpu)
 		cpufreq_stats_free_table(cpu);
 	cpufreq_allstats_free();
+	cpufreq_put_global_kobject();
 }
 
 MODULE_AUTHOR("Zou Nan hai <nanhai.zou@intel.com>");
-- 
1.9.1

